app/controllers/projects_controller.rb

```diff
@@ -31,6 +31,7 @@ def sort_stories
   def toggle_archive
     @project = Project.find(params[:id])
     @project.toggle_archived!
+    @project.reload
```

Comment:
hey yes @arielj , yes if we do something like this

```
   def archive
     Project.where(id: id).or(Project.where(parent_id: id)).update_all(status: "archived")
+    self.status = 'archived'
   end
 
   def unarchive
     Project.where(id: id).or(Project.where(parent_id: id)).update_all(status: nil)
+    self.status = nil
   end
```

It does work. Although I think it can be a little confusing to anyone in the future if they look at this method independently.

Another way to avoid DB query / reload, can maybe add some code in the controller itself, like
```
@project.status  = @project.status == nil ?  'archived' : nil
```
Although this also can be a bit confusing as to why we are explicitly setting up the value of `status`.


I would prefer to have the reload as it's the cleanest of the other possible options.
Let me know what do you think?

Reply:
got it @arielj . Thanks for explaining your thought process. :+1: 
I will make the change and push it again.

Reply:
I have pushed the changes @arielj